src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -42,7 +49,18 @@ internal ItemManager(RecentItemsManager recentItemsManager)
             IAsyncCompletionSession session,
             AsyncCompletionSessionInitialDataSnapshot data,
             CancellationToken cancellationToken)
-            => Task.FromResult(data.InitialList.OrderBy(i => i.SortText).ToImmutableArray());
+        {
+            if (session.TextView.Properties.TryGetProperty(CompletionSource.TargetTypeFilterExperimentEnabled, out bool isExperimentEnabled) && isExperimentEnabled)
+            {
+                // This method is called exactly once, so use the opportunity to set a baseline for telemetry.
+                if (data.InitialList.Any(i => i.Filters.Any(f => f.DisplayText == FeaturesResources.Target_type_matches)))
+                {
+                    Logger.Log(FunctionId.Intellisense_AsyncCompletion_SessionContainsTargetTypeFilter);
+                }
+            }
```

Comment:
I see the same code added in the CommitManager. Maybe we can extract it to a helper function. See Helpers.cs in the project.

Reply:
@ivanbasov Which bit exactly?

Reply:
I mean 
```
         if (session.TextView.Properties.TryGetProperty(CompletionSource.TargetTypeFilterExperimentEnabled, out bool isExperimentEnabled) && isExperimentEnabled)
            {
                // This method is called exactly once, so use the opportunity to set a baseline for telemetry.
                if (data.InitialList.Any(i => i.Filters.Any(f => f.DisplayText == FeaturesResources.Target_type_matches)))
                {
                    Logger.Log(FunctionId.Intellisense_AsyncCompletion_SessionContainsTargetTypeFilter);
                }
            }
```
can be extracted into a helper method.

Reply:
Discussed offline, we're going to leave as-is.